src/shared/common/Button.js

```diff
@@ -0,0 +1,68 @@
+import React, {Component} from 'react';
+import { Text, View, StyleSheet, Pressable} from 'react-native';
+
+class Button extends Component{
+  _renderChildren() {
+    let childElements = [];
+    React.Children.forEach(this.props.children, (item) => {
+      if (typeof item === 'string' || typeof item === 'number') {
+        const element = (
+          <Text
+            style={[styles.text, this.props.textStyle]}
+            key={item}>
+            {item}
+          </Text>
+        );
+        childElements.push(element);
+      } else if (React.isValidElement(item)) {
+        childElements.push(item);
+      }
+    });
+    return (childElements);
+  }
+  render() {
+    if (this.props.isDisabled === true) {
+      return (
+        <View style={[styles.button, this.props.style, styles.opacity]}>
+          {this._renderChildren()}
+        </View>
+      );
+    }
+    let extra_attributes = {testID: this.props.testID};
+
+    return (
+      <Pressable 
+        style={[styles.button, this.props.style]} 
+        onPress={this.props.onPress} 
+        android_ripple={{color:'light-gray'}}
+        {...extra_attributes}
+      >
+        {this._renderChildren()}
+      </Pressable>
+    )
+  };
+}
+
+const styles = StyleSheet.create({
```

Comment:
Can I suggest that we try to keep a consistent structure to component files?
I think for most of them, the stylesheet is at the top, maybe you can follow that same logic? It makes it much easier to navigate the codebase when we know where to find things :)
(there's no reason why top or bottom is better, it's purely standardization)